### PR TITLE
test: add container installer test

### PR DESCRIPTION
**Pull request by mvo5**
Add a new integration test that checks that the container installer
is working. The contaner installer just does an unattended install
of a disk. The test will run qemu with the installer.iso and an
empty disk. Once a reboot from the ISO is detected (via QMP) qemu
exists and boots the test disk and checks that it boots and the
test user can login.

### CODE REVIEW

File: test/test_smoke.py [hunk 1]
```diff
@@ -179,3 +179,20 @@ def test_image_build_without_se_linux_denials(image_type):
     assert image_type.journal_output != ""
     assert not log_has_osbuild_selinux_denials(image_type.journal_output), \
         f"denials in log {image_type.journal_output}"
+
+
+@pytest.mark.parametrize("image_type", ["iso"], indirect=["image_type"])
+def test_iso_installs(image_type):
+    installer_iso_path = image_type.img_path
+    test_disk_path = installer_iso_path.with_name("test-disk.img")
+    with open (test_disk_path, "w") as fp:
```
Comment (mvo5) [inline test/test_smoke.py:188]: This should probably move into the VM class as "create_disk" or something

File: test/vm.py [hunk 2]
```diff
@@ -38,23 +48,55 @@ def start(self):
...
+                self.wait_qmp_event(qmp_event)
+                self._log(f"qmp event {qmp_event}")
+            case _:
+                raise ValueError(f"unsupported wait_event {wait_event}")
 
     def _log(self, msg):
         # XXX: use a proper logger
         sys.stdout.write(msg.rstrip("\n") + "\n")
 
+    def wait_qmp_event(self, qmp_event):
```
Comment (mvo5) [inline test/vm.py:83]: Add wait_event_timeout here too (not just for the socket)?

**Comment by mvo5** [comment]
The testing farm tests fail with "no space left on device" :/

File: test/vm.py [hunk 3]
```diff
@@ -107,18 +125,52 @@ def start(self):
...
         self._qemu_p = subprocess.Popen(
-            qemu_cmdline, stdout=sys.stdout, stderr=sys.stderr)
+            qemu_cmdline,
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+        )
         # XXX: also check that qemu is working and did not crash
-        self.wait_ssh_ready()
-        self._log(f"vm ready at port {self._ssh_port}")
+        match wait_event.split(":"):
```
Comment (supakeen) [inline test/vm.py:146]: Do we need to run these tests downstream as well at some point? The `match`-statement was added to a quite recent Python.

Comment (lmilbaum) [reply test/vm.py:146]: As a general note, I would answer - yes.

Comment (mvo5) [reply test/vm.py:146]: Thank you, indeed `match` is only available in python 3.10 and centos9  has 3.9 and centos8 has 3.6 so I remove this.

Comment (mvo5) [reply test/vm.py]: This is done now

**Comment by bcl** [comment]
It looks like `SUPPORTED_IMAGE_TYPES` is no longer the list that drives all the test loops. I was a bit confused by the artifact test change until I realized that you have a separate test for `iso` that isn't in the list. Maybe it needs a new name, or a comment?

**Comment by mvo5** [comment]
> It looks like `SUPPORTED_IMAGE_TYPES` is no longer the list that drives all the test loops. I was a bit confused by the artifact test change until I realized that you have a separate test for `iso` that isn't in the list. Maybe it needs a new name, or a comment?

Thanks, that is a nice catch and a good point. I renamed the vars a bit to (hopefully) make it more clear. Ideas/comments welcome!

**Review by bcl — DISMISSED** [review]
Looks good to me, thanks!

**Comment by mvo5** [comment]
It looks like the test is currently failing because of an anaconda error?
Screenshot from 2024-01-25 10-28-39

**Review by achilleas-k — APPROVED** [review]
Splendid.  LGTM!

**Pull request merged**